Support executing SSE on ALM SaaS

Roy-Lu opened the pull request:
Support executing SSE on ALM SaaS
JIRA Ticket: JENKINS-42295

Commit:
Support executing SSE on ALM SaaS

Commit:
Remove the pattern check.

YafimK commented on src/main/java/com/hp/application/automation/tools/rest/RestClient.java:
```diff
@@ -4,13 +4,7 @@
-import java.net.PasswordAuthentication;
-import java.net.Proxy;
-import java.net.URL;
-import java.net.URLConnection;
```
Please avoid such changes - explicit import makes life easier to catch import issues

YafimK commented on src/main/java/com/hp/application/automation/tools/sse/sdk/RunManager.java:
```diff
@@ -5,11 +5,17 @@
+
 /**
  *
  * @author Effi Bar-She'an
```
Please add proper JavaDoc to this file

YafimK commented on src/main/java/com/hp/application/automation/tools/results/service/AlmRestTool.java:
```diff
@@ -17,7 +16,7 @@
-import com.hp.application.automation.tools.sse.sdk.RestAuthenticator;
+import com.hp.application.automation.tools.sse.sdk.authenticator.RestAuthenticator;
 
 public class AlmRestTool {
```
Please add JavaDoc

YafimK reviewed (changes requested):
Please add proper JavaDoc to the classes - at least some General info  on existing classes

Commit:
Support executing SSE on ALM SaaS
1. Fixed some check style issues.
2. Unify the rest authentication process here from separated part, ALMRestTool and RunManager.

YafimK commented:
approved for test build

Commit:
Fixed some lint issues.

YafimK commented:
approved for test build

Roy-Lu commented:
I think the hpebot is too strict.

Commit:
Fixed lint issues.

Roy-Lu commented:
MAJOR AuthenticationTool.java#L68: Either log or rethrow this exception. rule
It is logged.
MAJOR RestAuthenticator.java#L147: Remove this unused private "isAlreadyAuthenticated" method. rule
MAJOR RestAuthenticator.java#L171: Remove this unused private "logLoggedInSuccessfully" method. rule
MAJOR RestAuthenticatorSaas.java#L106: Remove this unused private "logLoggedInSuccessfully" method. 
These methods are used.

YafimK commented:
approve for test build
